module-6/src/businessLogic.service.ts

```diff
@@ -0,0 +1,40 @@
+// Or Service layer. The middle layer. All business logic related to our application might be in this layer. This layer knows about our application entities such as user, order, comment, post, and so on. It doesn't care what protocol is used for transferring data to a client, or whether data will be sent in a body or in a header.
+import { ProductEntity } from "./entities/product.entity.ts";
+import { getAllCarts, getAllProducts } from "./dataAccess.repository.ts";
+import { CartEntity } from "./entities/cart.entity.ts";
+import Joi from 'joi';
+
+export const findProductByID = (id: string): ProductEntity | undefined => {
+	const allProducts = getAllProducts();
+
+	const foundProduct = allProducts.find((el: ProductEntity) => {
+		return el.id === id;
+	});
+	return foundProduct;
```

Comment:
There is no need to declare a variable, simply return it.

```suggestion
	return allProducts.find((el: ProductEntity) => {
		return el.id === id;
	});
```